File: pickerkit/src/main/java/com/dewarder/pickerkit/result/PickerGalleryResult.java
package com.dewarder.pickerkit.result;

import com.dewarder.pickerkit.model.PickerMedia;
import com.dewarder.pickerkit.model.PickerMediaType;

import java.util.EnumSet;

public interface PickerGalleryResult extends PickerBaseMediaResult<PickerMedia> {

    PickerImageResult getImageResult();

    PickerVideoResult getVideoResult();

    EnumSet<PickerMediaType> getTypes();

    boolean containsType(PickerMediaType type);
}

File: pickerkit/src/main/java/com/dewarder/pickerkit/result/PickerGalleryResult.kt
package com.dewarder.pickerkit.result

import com.dewarder.pickerkit.core.Result
import com.dewarder.pickerkit.gallery.model.PickerMedia
import com.dewarder.pickerkit.gallery.model.PickerMediaType
import java.util.*

interface PickerGalleryResult : PickerBaseMediaResult<PickerMedia>, Result {

    fun getImageResult(): PickerImageResult

    fun getVideoResult(): PickerVideoResult

    val types: EnumSet<PickerMediaType>

    operator fun contains(type: PickerMediaType): Boolean
}
